sql-api/src/main/java/io/cdap/pipeline/sql/api/template/SQLNode.java

```diff
@@ -16,9 +16,14 @@
 
 package io.cdap.pipeline.sql.api.template;
 
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.tools.RelBuilder;
+
 /**
  * Represents an abstract SQL transformation node.
  */
-public abstract class SQLNode extends SQLConfigurable {
+public abstract class SQLNode {
```

Comment:
I think it's better for this to be an interface, since there is nothing related to implementation here

Reply:
Minor changes were needed for the joiner, so I decided to integrate the joiner interface a bit earlier. In this case, I turned SQLNode into an interface and created an SQLTransform abstract class instead.